### PR TITLE
Updated event push notification to have events unseen count. Updated group create/edit validation. Created getEvent handler.

### PR DESCRIPTION
**Pull request by john-andrews12**
## Summary
Read the title (To be more specific on the create/edit validation, the create did not have any validation before now and the event validation had the max group members constraint added to it).

## Testing
I deployed my changes to the ProxyJohn lambda and tested the app using my john api endpoint. I made sure the get event handler was working as intended.

I temporarily set the max group size down to 2 and made sure that I was unable to add a new user to a group/create a group with more than 2 users.

I created a new group.

Note I'm nearly certain the notification changes work, however I cannot be certain unless I update the EventResolver lambda which is a "prod" lambda.

### CODE REVIEW

File: backEnd/src/main/java/handlers/GetEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  private DbAccessManager dbAccessManager;
+  private Metrics metrics;
+
+  @Inject
+  public GetEventHandler(final DbAccessManager dbAccessManager, final Metrics metrics) {
+    this.dbAccessManager = dbAccessManager;
+    this.metrics = metrics;
+  }
+
+  public ResultStatus handle(final String activeUser, final String groupId, final String eventId) {
```
Comment (joshrap67) [inline backEnd/src/main/java/handlers/GetEventHandler.java:23]: Docs?

File: front_end_pocket_poll/lib/events_widgets/event_details_closed.dart [hunk 2]
```diff
@@ -196,8 +195,8 @@ class _EventDetailsClosedState extends State<EventDetailsClosed> {
     );
   }
 
-  void getEvent() {
-    this.event = Globals.currentGroupResponse.group.events[widget.eventId];
+  void getEvent(final Event event) {
```
Comment (joshrap67) [inline front_end_pocket_poll/lib/events_widgets/event_details_closed.dart:198]: I think this should be renamed now to something like "buildUserRows". I was very confused what this method was doing at first by calling it getEvent.

File: front_end_pocket_poll/lib/events_widgets/event_details_consider.dart [hunk 3]
```diff
@@ -313,8 +312,8 @@ class _EventDetailsConsiderState extends State<EventDetailsConsider> {
     }
   }
 
-  void getEvent() {
-    this.event = Globals.currentGroupResponse.group.events[widget.eventId];
+  void getEvent(final Event event) {
```
Comment (joshrap67) [inline front_end_pocket_poll/lib/events_widgets/event_details_consider.dart:315]: Same as above comment with renaming

File: front_end_pocket_poll/lib/events_widgets/event_details_occurring.dart [hunk 4]
```diff
@@ -200,8 +199,8 @@ class _EventDetailsOccurringState extends State<EventDetailsOccurring> {
     );
   }
 
-  void getEvent() {
-    this.event = Globals.currentGroupResponse.group.events[widget.eventId];
+  void getEvent(final Event event) {
```
Comment (joshrap67) [inline front_end_pocket_poll/lib/events_widgets/event_details_occurring.dart:202]: Same as above comment about renaming

File: front_end_pocket_poll/lib/events_widgets/event_details_voting.dart [hunk 5]
```diff
@@ -231,8 +230,8 @@ class _EventDetailsVotingState extends State<EventDetailsVoting> {
     );
   }
 
-  void getEvent() {
-    this.event = Globals.currentGroupResponse.group.events[widget.eventId];
+  void getEvent(final Event event) {
```
Comment (joshrap67) [inline front_end_pocket_poll/lib/events_widgets/event_details_voting.dart:233]: Same as above comment about renaming

File: front_end_pocket_poll/lib/widgets/members_page.dart [hunk 6]
```diff
@@ -306,6 +308,12 @@ class _MembersPageState extends State<MembersPage> {
           "Cannot add $username because they have previously left this group.",
           this.context);
       hideKeyboard(this.context);
+    } else if (widget.displayedMembers.length >= Globals.maxGroupMembers) {
+      showErrorMessage(
+          "Error",
+          "Cannot add $username because the max group size of ${Globals.maxGroupMembers} has been met.",
```
Comment (joshrap67) [inline front_end_pocket_poll/lib/widgets/members_page.dart:314]: Small request but could you put the username in quotes? Makes it easier to read imo and we do that elsewhere in the app.

File: front_end_pocket_poll/lib/widgets/members_page.dart [hunk 7]
```diff
@@ -335,21 +343,29 @@ class _MembersPageState extends State<MembersPage> {
...
-        }));
-    this
-        .displayedFavoritesRows
-        .removeWhere((row) => row.username == memberToAdd.username);
-    setState(() {});
+  void addMemberFromFavorites(final Favorite memberToAdd) {
+    if (widget.displayedMembers.length >= Globals.maxGroupMembers) {
+      showErrorMessage(
+          "Error",
+          "Cannot add ${memberToAdd.username} because the max group size of ${Globals.maxGroupMembers} has been met.",
```
Comment (joshrap67) [inline front_end_pocket_poll/lib/widgets/members_page.dart:350]: Same as above comments about quotes.

File: backEnd/src/main/java/handlers/GetEventHandler.java (hunk 1 above)
Comment (john-andrews12) [reply backEnd/src/main/java/handlers/GetEventHandler.java:23]: good call out

File: front_end_pocket_poll/lib/events_widgets/event_details_closed.dart (hunk 2 above)
Comment (john-andrews12) [reply front_end_pocket_poll/lib/events_widgets/event_details_closed.dart:198]: Oh right, I was just using what was already there with no thought to what it was actually doing after the update. I updated it to buildUserRows as suggested.

File: front_end_pocket_poll/lib/widgets/members_page.dart (hunk 6 above)
Comment (john-andrews12) [reply front_end_pocket_poll/lib/widgets/members_page.dart:314]: I was actually staying consistent with the other code on this page that does the same thing. I agree with you though so I'll update this and the old code.

**Pull request merged**